code/go/0chain.net/smartcontract/stakepool/lock.go

```diff
@@ -119,3 +119,33 @@ func (sp *StakePool) LockPool(
 
 	return nil
 }
+
+func (sp *StakePool) StakeForProvider(providerType spenum.Provider, providerID string, balances cstate.StateContextI) error {
+	staked, err := sp.stake()
```

Comment:
We should use the difference stakes between current total stake and the previous one, otherwise the provider's total stakes will be doubled each time when this function is called.  

Suppose a blobber now has 100 zcn total staked, a client staked 10 zcc to it, the total stake of the blobber will be 110, if the same client stake another 10, then the pool will have totally 20 zcn, which is the total staked value here, and after processing the total stake update event, the blobber's total stake will be 110+20 =  130, which is incorrect.

Reply:
@peterlimg it was simply moved from previous implementation,I wonder, how it was all working before. @Sriep do you know if it is all tested?

Reply:
I checked TotalOffers, doesn't look to be working correctly either
```
			tag, data := event.NewUpdateBlobberTotalOffersEvent(providerID, sp.TotalOffers)
```

```
func withBlobberTotalOffersAdded() eventMergeMiddleware {
	return withEventMerge(func(a, b *Blobber) (*Blobber, error) {
		a.OffersTotal += b.OffersTotal
		return a, nil
	})
}
```
looks like we also summarize total values here if several events are emitted, cmiiw @peterlimg

Reply:
changed for TotalStaked to be overwritten during merging

Reply:
@dabasov it worked previously because we use txn hash as pool id, so basically we would not stake to existing pool, the `sp.Stake()` is always a fresh one, and safe to be added to blobber's total stake.

Reply:
aaah, got  it, it's tricky